File: src/HMS.Domain/Entities/MedicalRecord.cs
using HMS.Domain.Common;
using HMS.Domain.Enums;

namespace HMS.Domain.Entities;

public class MedicalRecord : BaseEntity
{
    public Guid PatientId { get; set; }
    public Patient Patient { get; set; } = null!;

    public Guid DoctorId { get; set; }
    public Doctor Doctor { get; set; } = null!;

    public AdmissionType AdmissionType { get; set; }
    public DateTime VisitDate { get; set; } = DateTime.UtcNow;
    public string? ChiefComplaint { get; set; }
    public string? Diagnosis { get; set; }
    public string? TreatmentPlan { get; set; }
    public string? VitalSigns { get; set; }
    public string? Notes { get; set; }

    // Navigation
    public ICollection<Prescription> Prescriptions { get; set; } = new List<Prescription>();
    public ICollection<LabTest> LabTests { get; set; } = new List<LabTest>();
}

File: frontend/src/types/medicalRecord.ts
export enum AdmissionType {
  OPD = 1,
  IPD = 2,
}

export const ADMISSION_TYPE_LABELS: Record<AdmissionType, string> = {
  [AdmissionType.OPD]: "OPD",
  [AdmissionType.IPD]: "IPD",
};

export interface MedicalRecordDto {
  id: string;
  patientId: string;
  patientName: string;
  doctorId: string;
  doctorName: string;
  specialization: string;
  admissionType: AdmissionType;
  visitDate: string;
  chiefComplaint: string | null;
  diagnosis: string | null;
  treatmentPlan: string | null;
  vitalSigns: string | null;
  notes: string | null;
  createdAt: string;
}

export interface CreateMedicalRecordDto {
  patientId: string;
  doctorId: string;
  admissionType: AdmissionType;
  visitDate: string;
  chiefComplaint?: string;
  diagnosis?: string;
  treatmentPlan?: string;
  vitalSigns?: string;
  notes?: string;
}

export interface UpdateMedicalRecordDto {
  admissionType: AdmissionType;
  visitDate: string;
  chiefComplaint?: string;
  diagnosis?: string;
  treatmentPlan?: string;
  vitalSigns?: string;
  notes?: string;
}
